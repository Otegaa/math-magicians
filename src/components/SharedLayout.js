import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { Outlet } from 'react-router-dom';
import Navbar from './Navbar';

const SharedLayout = () => (
  <>
    <Navbar />
    <Outlet />
  </>
);

export default SharedLayout;
