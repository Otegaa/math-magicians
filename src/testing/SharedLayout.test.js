// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import SharedLayout from '../components/SharedLayout';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <SharedLayout />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
