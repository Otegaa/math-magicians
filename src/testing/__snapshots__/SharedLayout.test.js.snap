// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "space-between",
      "margin": "8rem auto",
      "maxWidth": "80%",
      "padding": "1rem",
      "width": "50%",
    }
  }
>
  <h1>
    <a
      aria-current="page"
      className="heading active"
      href="/"
      onClick={[Function]}
    >
      Math magicians
    </a>
  </h1>
  <nav
    style={
      Object {
        "display": "flex",
        "fontSize": "1.7rem",
        "gap": "1rem",
      }
    }
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "color": "red",
        }
      }
    >
      Home
    </a>
    <span>
      |
    </span>
    <a
      className=""
      href="/calculator"
      onClick={[Function]}
      style={
        Object {
          "color": "grey",
        }
      }
    >
      Calculator
    </a>
    <span>
      |
    </span>
    <a
      className=""
      href="/quotes"
      onClick={[Function]}
      style={
        Object {
          "color": "grey",
        }
      }
    >
      Quotes
    </a>
  </nav>
</div>
`;
